### ISSUE TITLE
ytt does not support a directory passed to data-values-file

### ISSUE THREAD

**Issue by itmustbejj** [k14s/ytt#753]
**What steps did you take:**
```
➜ ./examples/data-values-directory/run.sh
```

**What happened:**
```
ytt: Error: Extracting data value from file:
  Reading file 'examples/data-values-directory/values/'
```
**What did you expect:**
I expected ytt to render correctly and the output to match the example's expected.txt

**Anything else you would like to add:**
Appears to be a regression, since this functionality was added in https://github.com/vmware-tanzu/carvel-ytt/pull/670

**Environment:**
- ytt version 0.36.0
---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by mamachanko** [comment]
@itmustbejj thank you for reporting this!

#670 was released with [v0.41.0](https://github.com/vmware-tanzu/carvel-ytt/releases/tag/v0.41.0).

I can confirm that the example works with the latest version of `ytt`:
```shell
❯ ./examples/data-values-directory/run.sh
values:
  name: suggestion-service
  instances: 4
  accept_insecure_conns: false
  cache:
    driver: redis
    config:
      maxEntries: 1024
      strategy: MRU
      host: localhost:6379
      tls-client-cert-file: client.crt
      tls-client-key-file: client.key

❯ ytt version
ytt version 0.43.0
```

In other words, the feature isn't yet available for the version you have reported. The example did not even exist at that time:
```shell
❯ git co v0.36.0
Note: switching to 'v0.36.0'.

# ...

❯ ./examples/data-values-directory/run.sh
-bash: ./examples/data-values-directory/run.sh: No such file or directory
```

If you are able to upgrade to the latest version that would be optimal. If your environment is constrained to `v0.36.0` - b/c maybe you are depending on a kapp-controller version that you do not control - then, unfortunately, you must approach your values files in a `v0.36.0`-compliant way. If you need help with that we would be happy to assist you here or in the [kubernetes#carvel](https://kubernetes.slack.com/archives/CH8KCCKA5) Slack channel.

**Comment by pivotaljohn** [comment]
Thank you, @mamachanko!

@itmustbejj we're going to presume this resolved your concerns (and we'll close this issue). If it didn't, please do comment on _this_ issue and it'll bump back on our radar. 👍🏻
